src/copaw/agents/model_factory.py

```
@@ -118,13 +118,51 @@ class FileBlockSupportFormatter(base_formatter_class):
+                    logger.warning(
+                        "Assistant message count mismatch after formatting "
+                        "(%d before, %d after). "
+                        "Skipping reasoning_content injection.",
+                        len(in_assistant),
+                        len(out_assistant),
+                    )
+                else:
+                    for in_msg, out_msg in zip(
+                        in_assistant,
+                        out_assistant,
+                    ):
```

Ruff (0.15.2)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)